accepted/single-file/design.md

```diff
@@ -66,225 +32,254 @@ Publishing to a single file can be triggered by adding the following property to
+      ```
+  
+    * The `bundle_probe` function pointer is passed to the runtime (encoded as a string) through a property named `BUNDLE_PROBE`.
+  
+    * The contents of the bundle are not included in `TRUSTED_PLATFORM_ASSEMBLIES`. Any additional assemblies on disk (ex: due to `additional-deps` /`AdditionalProbingPaths`) are included in `TRUSTED_PLATFORM_ASSEMBLIES`.
+  
+### Host Builds
+
+The .net 5 AppHost will be built in a few different configurations, and consumed by the SDK in appropriate scenarios as noted below.
+
+| Code Name  | Build                                                        | Scenario                                                     |
+| ---------- | ------------------------------------------------------------ | ------------------------------------------------------------ |
+| AppHost    | Current AppHost build                                        | All non-single-file apps<br />Framework-dependent single-file apps |
+| StaticHost | AppHost with HostFxr and HostPolicy statically linked.       | Self-contained single-file apps on Windows                   |
+| SuperHost  | StaticHost with CoreCLR runtime components statically linked | Self-contained single-file apps on Unix systems              |
+
+Ideally, we should use the SuperHost for self-contained single-file apps on Windows too. However, due to certain limitations in debugging experience and the ability to collect Watson dumps, etc., the CoreCLR libraries are left on disk beside the app.
+
+The files typically published for self-contained apps on Windows are:
+
+* The `App.Exe`-- StaticHost along with assemblies and configuration files bundled as a single-file app.
+* `coreclr.dll`(runtime), `clrjit.dll` (JIT compiler), and `clrcompression.dll` (native counterpart of BCL for certain compression algorithms).
+    * It may be possible to link these DLLs together as CoreCLR.dll, but this work is not prioritized, in deference to building the full super-host once debugging framework supports it.
+* mscordaccore.dll (to enable Watson dumps)
+
+Certain additional binaries may be optionally included with the app to enable debugging scenarios. 
+The work to not require these binaries as separate files on disk alongside the app is ongoing.
+
+* For F5 debugging on VS and VS-Core: `mscordbi.dll`, `libmscordbi.so`
+* For Linux mini-dumps: `createdump`, `libmscordaccore.so` 
+* For ETW / LTTng: `clretwrc.dll`, `libcoreclrtraceptprovider.so` 
+* For exception stack trace source information, error string resources on Windows: `Microsoft.DiaSymReader.Native.amd64.dll`, `mscorrc.debug.dll`, `mscorrc.dll`
+
+When targeting `win7` platform, several additional DLLs are necessary (`api-*.dll`) to handle API compatibility. These files must be alongside the AppHost for the app to even start execution.
+Therefore, we propose that targeting `win7-*` should not be supported when publishing apps as a single-file.
```

Comment:
@richlander - What is the decision for Win7?

Reply:
@samsp-msft  @richlander To state the constraint more explicitly, if we do support win7 targets, then the self-contained single-file publish for hello-world console app will be 47 files, instead of 224 files. These files cannot be bundled within the app and later extracted. Please see: https://github.com/dotnet/runtime/issues/13356#issuecomment-530595570

Reply:
I'm pretty sure we should be able to build the apphost as a Win7 compatible executable without the additional dlls (after all Win10 can run Win7 exes just fine). The question is if that has other side effects - like increasing the size of the apphost, or turning on compatibility shims on Win10 and so on. In the extreme we could have a Win7 specific apphost.